### PR TITLE
feat: Specify Managed Prefix List for access control

### PR DESCRIPTION
**Pull request by yo-ga**
### Issue

- Resolves kubernetes-sigs/aws-load-balancer-controller#2167

### Description

- Add new annotations for Managed Prefix List
- Combine the IP addresses and Managed Prefix Lists
  - User can add IP addresses and Managed Prefix Lists at the same time

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [x] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/service/model_build_managed_sg.go [hunk 1]
```diff
@@ -113,6 +129,9 @@ func (t *defaultModelBuildTask) buildCIDRsFromSourceRanges(_ context.Context, ip
 		}
 	}
 	if len(cidrs) == 0 {
+		if prefixListsConfigured {
```
Comment (oliviassss) [inline pkg/service/model_build_managed_sg.go:132]: just for my understanding - why we allow an empty cidr, if prefixLists is configured?

File: pkg/service/model_build_managed_sg_test.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+				},
+			},
+		},
+		{
+			name: "no ip range but prefix list",
+			fields: fields{
+				svc: &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"service.beta.kubernetes.io/aws-load-balancer-security-group-prefix-lists": "pl-xxxxx",
```
Comment (oliviassss) [inline pkg/service/model_build_managed_sg_test.go:195]: nit: can we test against multiple lists too?

**Comment by oliviassss** [comment]
@yo-ga, sorry for late response, and thanks for your contribution. I left some comments.

File: pkg/service/model_build_managed_sg.go (hunk 1 above)
Comment (yo-ga) [reply pkg/service/model_build_managed_sg.go:132]: If the user just wants to allow the traffic only from the prefix list, the controller needs to skip the unlimited CIDR settings (`0.0.0.0/0` or `::/0`)

File: docs/guide/service/annotations.md [hunk 3]
```diff
@@ -437,6 +438,22 @@ Load balancer access can be controlled via following annotations:
...
+
+    !!!warning ""
+        If you'd like to use this annotation, make sure your security group rule quota is enough. If you'd like to know how the managed prefix list affects your quota, see the [reference](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-aws-managed-prefix-lists.html#aws-managed-prefix-list-weights) in the AWS documentation for more details.
+
+    !!!tip ""
+        If you only use this annotation without `load-balancer-source-ranges`, the controller managed security group would ignore the `load-balancer-source-ranges` default settings.
+
+    !!!example
+        ```
+        service.beta.kubernetes.io/aws-load-balancer-security-group-prefix-lists: pl-00000000
```
[the remark this answers is not in the document]
Comment (yo-ga) [reply docs/guide/service/annotations.md]: Thanks for the suggestion. Updated in [66afd13](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3584/commits/66afd137ca25b5faf7aeea8508f305ace8e4181c)

File: pkg/service/model_build_managed_sg_test.go (hunk 2 above)
Comment (yo-ga) [reply pkg/service/model_build_managed_sg_test.go:195]: The test is added in [51c5998](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3584/commits/51c59981db512ff83c7c1297b7e7a597fac7d6ad)

**Comment by yo-ga** [comment]
Hi @oliviassss , kindly confirm the change and the reply

**Comment by yo-ga** [comment]
Hi @oliviassss, the test seems well.

**Comment by ignatenkobrain** [comment]
@M00nF1sh can you check / approve / merge please?

**Comment by bohuslav-broza** [comment]
@M00nF1sh hello, any update on this? Thank you

**Comment by yo-ga** [comment]
Hi @M00nF1sh , one more thing to ask, when will the feature be released?

**Comment by dcodix** [comment]
Hi @M00nF1sh , same question as @yo-ga , is there an estimation on when will this be released?

**Comment by dcodix** [comment]
I just tested v2.8.0 and it seems to be working. Thanks!

**Pull request merged**